src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -1780,48 +1778,51 @@ private bool PrepareToPushMessageToBridge<T>(Message message, ResultProcessor<T>
             return false;
         }
 
-        private ValueTask<WriteResult> TryPushMessageToBridgeAsync<T>(Message message, ResultProcessor<T> processor, IResultBox<T> resultBox, ref ServerEndPoint server)
+        private ValueTask<WriteResult> TryPushMessageToBridgeAsync<T>(Message message, ResultProcessor<T>? processor, IResultBox<T>? resultBox, [NotNullWhen(true)] ref ServerEndPoint? server)
             => PrepareToPushMessageToBridge(message, processor, resultBox, ref server) ? server.TryWriteAsync(message) : new ValueTask<WriteResult>(WriteResult.NoConnectionAvailable);
 
         [Obsolete("prefer async")]
-        private WriteResult TryPushMessageToBridgeSync<T>(Message message, ResultProcessor<T> processor, IResultBox<T> resultBox, ref ServerEndPoint server)
+        private WriteResult TryPushMessageToBridgeSync<T>(Message message, ResultProcessor<T>? processor, IResultBox<T>? resultBox, [NotNullWhen(true)] ref ServerEndPoint? server)
             => PrepareToPushMessageToBridge(message, processor, resultBox, ref server) ? server.TryWriteSync(message) : WriteResult.NoConnectionAvailable;
 
         /// <summary>
         /// Gets the client name for this multiplexer.
         /// </summary>
         public override string ToString() => string.IsNullOrWhiteSpace(ClientName) ? GetType().Name : ClientName;
 
-        internal Exception GetException(WriteResult result, Message message, ServerEndPoint server) => result switch
+        internal Exception GetException(WriteResult result, Message message, ServerEndPoint? server) => result switch
         {
-            WriteResult.Success => null,
+            WriteResult.Success => throw new ArgumentOutOfRangeException(nameof(result), "Be sure to check result isn't successful before calling GetException."),
```

Comment:
The intent is to throw here - it should never be hit. Overall: these are wrapped in a cheaper (than stack call) `result != WriteResult.Success`, _not_ doing that goes kaboom. We want to enforce code patterns of the cheap check, and not doing `.GetException(...)!` everywhere since we use the result.